### PR TITLE
feature (refs dplan-11493): add new event Interface

### PR DESCRIPTION
**Pull request by jalalifardemos**
**Ticket** :  https://demoseurope.youtrack.cloud/issue/DPLAN-11493/Uber-E-Mail-Import-eingereichte-STN-konnen-nicht-geloscht-werden

https://github.com/demos-europe/demosplan-core/pull/3014
https://github.com/demos-europe/demosplan-addon-maillane/pull/114

Description: Add new Interface for DeleteEmailImportedStatementEvent

### CODE REVIEW

**Review by bakridemos — CHANGES_REQUESTED** [review]

File: src/Contracts/Events/DeleteEmailImportedStatementEventInterface.php [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?php
+
+namespace DemosEurope\DemosplanAddon\Contracts\Events;
+
+interface DeleteEmailImportedStatementEventInterface
+{
+
```
[the remark this answers is not in the document]
Comment (jalalifardemos) [reply src/Contracts/Events/DeleteEmailImportedStatementEventInterface.php]: done in [79e93e4](https://github.com/demos-europe/demosplan-addon/pull/107/commits/79e93e4053465f20d712c61b284104de72d8c167)

File: src/Contracts/Events/DeleteEmailImportedStatementEventInterface.php [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<?php
+
+namespace DemosEurope\DemosplanAddon\Contracts\Events;
+
+use DemosEurope\DemosplanAddon\Contracts\Entities\StatementInterface;
+
+interface DeleteEmailImportedStatementEventInterface
```
Comment (mussbach) [inline src/Contracts/Events/DeleteEmailImportedStatementEventInterface.php:7]: I would introduce a generic `StatementPreDeleteEvent` that can be used for multiple purposes

**Review by mussbach — CHANGES_REQUESTED** [review]

File: src/Contracts/Events/DeleteEmailImportedStatementEventInterface.php (hunk 2 above)
Comment (jalalifardemos) [reply src/Contracts/Events/DeleteEmailImportedStatementEventInterface.php]: done in [7d93eca](https://github.com/demos-europe/demosplan-addon/pull/107/commits/7d93eca1abbf759f982509cc31c93e2d42ddcf1b)

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,6 +1,7 @@
 # Changelog
 
 ## UNRELEASED
+- add interface for new event DeleteEmailImportedStatementEvent
```
Comment (mussbach) [inline CHANGELOG.md:4]:
```suggestion
- add interface for new event StatementPreDeleteEventInterface
```

**Review by mussbach — CHANGES_REQUESTED** [review]
one minor remark

File: CHANGELOG.md (hunk 3 above)
Comment (jalalifardemos) [reply CHANGELOG.md]: done in [d70cb84](https://github.com/demos-europe/demosplan-addon/pull/107/commits/d70cb84f32c8d13bfce0d927bf7bff8640c77397)

**Review by mussbach — APPROVED** [review]
lgtm

**Review by bakridemos — APPROVED** [review]
lgtm

**Pull request merged**